pkg/ottl/expression.go

```diff
@@ -245,6 +246,53 @@ func (g StandardFloatGetter[K]) Get(ctx context.Context, tCtx K) (float64, error
 	}
 }
 
+// FunctionGetter uses a function factory to return an instantiated function as an Expr.
+type FunctionGetter[K any] interface {
+	Get(args Arguments) (Expr[K], error)
+}
+
+// StandardFunctionGetter is a basic implementation of FunctionGetter
+type StandardFunctionGetter[K any] struct {
+	fCtx FunctionContext
+	fact Factory[K]
+}
+
+// Get takes an Arguments struct containing arguments the caller wants passed to the
+// function and instantiates the function with those arguments.
+// If there is a mismatch between the function's signature and the arguments the caller
+// wants to pass to the function, an error is returned.
+func (g StandardFunctionGetter[K]) Get(args Arguments) (Expr[K], error) {
+	if g.fact == nil {
+		return Expr[K]{}, fmt.Errorf("undefined function")
+	}
+	fArgs := g.fact.CreateDefaultArguments()
+	if reflect.TypeOf(fArgs).Kind() != reflect.Pointer || reflect.TypeOf(args).Kind() != reflect.Pointer {
```

Comment:
We should check for each of these separately, the resolution for the error is different in each case.

Reply:
Checking these separately with different error messages.